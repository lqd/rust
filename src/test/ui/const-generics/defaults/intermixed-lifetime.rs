// revisions: full min
// Checks that lifetimes cannot be interspersed between consts and types.
#![cfg_attr(full, feature(const_generics))]
#![cfg_attr(full, allow(incomplete_features))]

struct Foo<const N: usize, 'a, T = u32>(&'a (), T);
//~^ Error lifetime parameters must be declared prior to const parameters
//[min]~^^ Error using type defaults and const parameters in the same parameter list

struct Bar<const N: usize, T = u32, 'a>(&'a (), T);
//[full]~^ Error lifetime parameters must be declared prior to defaulted parameters
//[min]~^^ Error using type defaults and const parameters in the same parameter list
//[min]~| Error lifetime parameters must be declared prior to defaulted parameters

fn main() {}
