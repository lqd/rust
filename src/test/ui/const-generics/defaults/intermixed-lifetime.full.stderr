error: lifetime parameters must be declared prior to const parameters
  --> $DIR/intermixed-lifetime.rs:6:28
   |
LL | struct Foo<const N: usize, 'a, T = u32>(&'a (), T);
   |           -----------------^^---------- help: reorder the parameters: `<'a, const N: usize, T = u32>`
   |
   = note: the expected order for parameters is: lifetimes, then consts and types, then defaulted parameters

error: lifetime parameters must be declared prior to defaulted parameters
  --> $DIR/intermixed-lifetime.rs:10:37
   |
LL | struct Bar<const N: usize, T = u32, 'a>(&'a (), T);
   |           --------------------------^^- help: reorder the parameters: `<'a, const N: usize, T = u32>`
   |
   = note: the expected order for parameters is: lifetimes, then consts and types, then defaulted parameters

error: aborting due to 2 previous errors

